Remove sensors from component category and represent the separately

tiyash-basu-frequenz opened the pull request:
This update introduces a dedicated RPC method for listing sensors in the microgrid, separating them from the "component" category. This new method allows for optional sensor filtering by ID or category. The change clarifies the term "component" to exclusively refer to electrical equipment like inverters and batteries, while sensors are now treated as a distinct entity.

tiyash-basu-frequenz committed:
Refactor the microgrid.proto file

The messages have been organised in a more logical way, in the following
order:
* components
* connections
* parameters for setting power levels
* parameters for setting bounds
* parameters for setting component states

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Separate sensors from ComponentCategory with new RPC methods

The change aims to clarify the distinction between electrical components
(e.g., inverters, batteries) and sensors (e.g., temperature, humidity).

This commit introduces a new RPC method for listing sensors in the
microgrid, effectively separating them from the "component" category.
This allows for more granular control and filtering of sensors by ID and
category. Additionally, a new RPC method for subscribing to sensor data
is introduced.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Remove obsolete message `PowerLevelParam`

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

thomas-nicolai-frequenz reviewed (dismissed):
All good from my side. We should however, clarify that when we say components we mean electrical components!

tiyash-basu-frequenz committed:
Update RPC documentation

In this change, componentts are re-inforced as electrical components,
in the light of the new sensor RPCs.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz commented:
>We should however, clarify that when we say components we mean electrical components!

@thomas-nicolai-frequenz I added another commit to re-inforce components as electrical components. Ready for another look.

stefan-brus-frequenz commented on proto/frequenz/api/microgrid/microgrid.proto:
```diff
@@ -490,3 +512,61 @@ message Connection {
+  frequenz.api.common.v1.metrics.Bounds bounds = 3;
+}
+
+// Encapsulation of a component ID, intended to be used as a parameter for rpc
```
There are plenty of new `component_id` fields introduced here with the type `uint64`. Should these be `ComponentIDParam`? If not, I think the docs here (and for sensor below) should be clarified that these are only intended to be used as top-level parameters for the rpc methods themselves, and perhaps a little blurb explaining why.

tiyash-basu-frequenz replied:
This will go away once https://github.com/frequenz-floss/frequenz-api-microgrid/issues/149 is done. Is it okay if I handle this directly there?

stefan-brus-frequenz reviewed (commented):
Looks pretty good overall, with one reservation
